Minor fixes

Anaphory opened the pull request:
This pull request suggests three minor fixes:

 - Do not catch exceptions when Beastling is run from `python -m`, in order to make debugging easier
 - Move some binary model properties up in the type hierarchy, without actually changing much
 - Fix one minor issue in the documentation

If I find the opportunity to add tests or other documentation fixes (eg. #202 #209)

This is a duplicate of #210, except now I picked the correct base branch here.

Commit:
Fix bit in documentation

Commit:
Improve binary model infrastructure.

Move the share_parameters handling to a subclass of BinaryModel.
That class provides a method which is either the model's name or the list of all
qualified feature names, depending on whether the model shares parameters or not.

Commit:
Wrap errors only when called from the command line.

Do not catch-and-sys.exit errors when called from `python -m`, so that the it is
easier to debug (using pdb.pm(), eg.) errors from calling Beastling from the
command line.

This is achieved by defining which errors should be caught and turned into
various exit codes using `sys.exit()` – the default is that all exceptions be
caught, but `__main__` re-defines that variable to a class that is never raised,
and therefore does not catch most exceptions.

I am sure there is a better way to run this, and I would be happy to see it
implemented, but I have no intuition what it might be.

Commit:
Add test for (1) bad share_params value and (2) covarion fine probs.

(1) This is one of the issues listed in #202.
(1) and (2) are missed lines in
https://codecov.io/gh/lmaurits/BEASTling/compare/2960b0290f125ec90dfb66a5f61926b2bb7936a6...4896606e08e0542bbd7ffb337f5a8712c11b0911/diff

Anaphory commented:
I don't know how I would reasonably test the `__main__` error handling procedure. Any suggenstions, @lmaurits  and @xrotwang, or may I merge this as-is?

xrotwang commented:
AFAICT, the only course of action taken by BEASTling upon encountering the new exception class, is terminating execution. So I guess, no tests are required, there either will be regular script termination or an error will be thrown - and the script will terminate.

Anaphory commented:
The idea is that this new `Exception` subclass is different from existing subclasses, so errors in

    try:
       ...
    except wrap_errors as e:
        errmsg("Error encountered while parsing configuration file:\n")
        traceback.print_exc()
        sys.exit(2)

are only caught when `cli.wrap_errors` has its default value `Exception`, not when `__main__` came along and redefined `cli.wrap_errors`, but the result (termination, which is the desired effect) is exactly as you describe. (`NoneException` will not be raised.)

Commit:
Separate fine probabilites test out into separate function

This means that we can later check the logfile to see whether it actually
contains ‘fine_probs’.
